finetune/adapter.py

```diff
@@ -47,18 +48,36 @@
 }
 
 
-def main(
+def setup(
     data_dir: Path = Path("data/alpaca"),
     checkpoint_dir: Path = Path("checkpoints/stabilityai/stablelm-base-alpha-3b"),
     out_dir: Path = Path("out/adapter/alpaca"),
     precision: Literal["bf16-true", "32-true"] = "bf16-true",
+    tpu: bool = False
 ):
-    check_valid_checkpoint_dir(checkpoint_dir)
+    strategy = "auto" if devices <= 1 else \
+        (XLAStrategy(sync_module_states=False)) if tpu \
+        else DeepSpeedStrategy(config=ds_config)
 
     fabric = L.Fabric(
-        devices=devices, strategy=(DeepSpeedStrategy(config=ds_config) if devices > 1 else "auto"), precision=precision
+        devices=devices, strategy=strategy, precision="32-true" if tpu else precision
     )
-    fabric.launch()
+    
+    if tpu:
+        p_main = functools.partial(main, data_dir, checkpoint_dir, out_dir)
+        fabric.launch(p_main)
+    else:
+        fabric.launch()
+        main(data_dir, checkpoint_dir, out_dir, fabric)
```

Comment:
We can use the launched function pattern for both cases

```suggestion
    fabric.launch(main, data_dir, checkpoint_dir, out_dir)
```